lms/djangoapps/student_account/test/test_views.py

```diff
@@ -697,24 +697,6 @@ def test_order_history_with_no_product(self):
 
         self.assertEqual(len(order_detail), 1)
 
-    def test_order_history_with_coupon(self):
-        """
-        Verify that get_order_details returns empty receipt_url for coupon product.
-        """
-        response = {
-            'results': [
-                factories.OrderFactory(
-                    lines=[
-                        factories.OrderLineFactory(product=factories.ProductFactory(product_class='Coupon'))
-                    ]
-                )
-            ]
-        }
-        with mock_get_orders(response=response):
-            order_detail = get_user_orders(self.user)
-
-        self.assertEqual(order_detail[0]['receipt_url'], '')
```

Comment:
I did that. https://github.com/edx/edx-platform/pull/14943/files#diff-9d6a9cfad4c03291e1a7ed78933c25abR186 the rest is done in another test case.